sw/device/silicon_creator/lib/drivers/flash_ctrl.h

```diff
@@ -202,85 +202,104 @@ typedef struct flash_ctrl_status {
 void flash_ctrl_status_get(flash_ctrl_status_t *status);
 
 /**
- * Perform a read transaction.
+ * Reads data from the data partition.
  *
  * The flash controller will truncate to the closest, lower word aligned
  * address. For example, if 0x13 is supplied, the controller will perform a read
  * at address 0x10.
  *
- * On success, `data` is populated with the read data.
+ * @param addr Address to read from.
+ * @param word_count Number of bus words to read.
+ * @param[out] data Buffer to store the read data.
+ * @return Result of the operation.
+ */
+rom_error_t flash_ctrl_data_read(uint32_t addr, uint32_t word_count,
+                                 uint32_t *data);
```

Comment:
Is there a reason that this takes (length, ptr) and the other takes (ptr, length)?

Reply:
Not really, I think I prefer buf, len order, but out param is at the end for this one.

Reply:
I think at some point the style guides preferred output parameters at the end of the function.

Reply:
Resolving this to unblock other PRs. PLMK if you have a strong preference either way.

Reply:
There is a conflict, so I'll update this too to use length, ptr for the sake of consistency.